integration-cli/docker_cli_ps_test.go

```diff
@@ -554,3 +554,44 @@ func (s *DockerSuite) TestPsFormatHeaders(c *check.C) {
 		c.Fatalf(`Expected 'NAMES\ntest\n', got %v`, out)
 	}
 }
+
+// Test for GitHub issue #12595
+func (s *DockerSuite) TestPsImageIDAfterUpdate(c *check.C) {
+
+	originalImageName := "busybox:TestPsImageIDAfterUpdate-original"
+	updatedImageName := "busybox:TestPsImageIDAfterUpdate-updated"
+
+	runCmd := exec.Command(dockerBinary, "tag", "busybox:latest", originalImageName)
+	out, _, err := runCommandWithOutput(runCmd)
+	c.Assert(err, check.IsNil)
+
+	originalImageID, err := getIDByName(originalImageName)
+	c.Assert(err, check.IsNil)
+
+	runCmd = exec.Command(dockerBinary, "run", "-d", originalImageName, "top")
+	out, _, err = runCommandWithOutput(runCmd)
+	c.Assert(err, check.IsNil)
+	containerID := strings.TrimSpace(out)
+
+	runCmd = exec.Command(dockerBinary, "commit", containerID, updatedImageName)
+	out, _, err = runCommandWithOutput(runCmd)
+	c.Assert(err, check.IsNil)
+
+	runCmd = exec.Command(dockerBinary, "tag", "-f", updatedImageName, originalImageName)
+	out, _, err = runCommandWithOutput(runCmd)
+	c.Assert(err, check.IsNil)
+
+	linesOut, err := exec.Command(dockerBinary, "ps").CombinedOutput()
```

Comment:
Let's actually do a --no-trunc here, because there's currently a bug where the full image id is shown but it should only show up to 12 chars... or in your assert below do `originalImageID[12]`